Sources/SwiftParser/SyntaxUtils.swift

```diff
@@ -41,6 +43,47 @@ extension RawUnexpectedNodesSyntax {
   }
 }
 
+// MARK: Combining unexpected nodes
+
+/// Implementation detail of `RawUnexpectedNodesSyntax.init(combining:arena)`.
+protocol UnexpectedNodsCombinable {
+  var elements: [RawSyntax] { get }
+}
+
+extension Array: UnexpectedNodsCombinable where Element: UnexpectedNodsCombinable {
+  var elements: [RawSyntax] { self.flatMap { $0.elements } }
+}
+
+extension Optional: UnexpectedNodsCombinable where Wrapped: UnexpectedNodsCombinable {
+  var elements: [RawSyntax] {
+    if let self = self {
+      return self.elements
+    } else {
+      return []
+    }
+  }
+}
+
+extension RawTokenSyntax: UnexpectedNodsCombinable {
+  var elements: [RawSyntax] {
+    return [RawSyntax(self)]
+  }
+}
+
+extension RawUnexpectedNodesSyntax: UnexpectedNodsCombinable {}
+
+extension RawUnexpectedNodesSyntax {
+  init?<T1: UnexpectedNodsCombinable, T2: UnexpectedNodsCombinable>(combining syntax1: T1, _ syntax2: T2, arena: SyntaxArena) {
```

Comment:
I don’t think so because then all nodes would need to be of the same type and you can’t combine e.g. `RawUnexpectedNodesSyntax` and `RawTokenSyntax`.

The following also doesn’t compile:

```swift
func foo<T>(_ x: T...) {}
foo(1, "x") // error: Conflicting arguments to generic parameter 'T' ('String' vs. 'Int')
```

Reply:
Need us some variadic generics :(

Reply:
Yep